storage/replica_raftstorage.go

```diff
@@ -112,16 +114,32 @@ func (r *Replica) Entries(lo, hi, maxBytes uint64) ([]raftpb.Entry, error) {
 	// If neither the number of entries nor the size limitations had an
 	// effect, we weren't able to supply everything the client wanted.
 	if len(ents) != int(hi-lo) && (maxBytes == 0 || size < maxBytes) {
```

Comment:
I'll add some tests around maxbytes to make sure it returns the correct values, even if maxbytes is hit and add the check for gaps/compacted into the scan.  .
And if there are gaps, I'll return ErrUnavailable  .